Enhance client creation - add `quarkus.minio.allow-empty` property

JiriOndrusek opened the pull request:
fixes https://github.com/quarkiverse/quarkus-minio/issues/64

PR adds property `quarkus.minio.allowEmpty` + junit test + doc update.


 If `quarkus.minio.allowEmpty` is set to `true` and `quarkus.minio.url` is not valid, producer returns `null` instead of throwing `ConfigurationException`. 

@jtama I changed scope of the client to be `@Dependent`, because it could return `null`.

Default behavior is not changed.

jtama commented:
I think `allowEmpty` should not let invalid configuration pass, only empty ones.

JiriOndrusek commented:
oh my bad, it allows only empty one - I'll rephrase docs and comments

JiriOndrusek commented:
@jtama I fixed comments and doc. I also added:

- One test covering behavior when options are valid + allow empty is `true` 
- One test where  options are invalid + allow empty is `true`

, to make sure, that allowEmpty works only if other options are not set.

jtama commented:
@all-contributors please add @JiriOndrusek for code
